internal/project/golang/toolchain.go

```diff
@@ -222,6 +218,9 @@ func (toolchain *Toolchain) CompileDockerfile(output *dockerfile.Output) error {
 		base.Step(step.Copy("./"+file, "./"+file))
 	}
 
+	base.Step(step.Run("go", "mod", "download").MountCache(filepath.Join(toolchain.meta.GoPath, "pkg"))).
```

Comment:
instead, we should copy in all `go.mod`, `go.sum` and run `go mod download` in each canonicalPath